client/src/ingredients/browser-compatibility-table/index.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from "react";
+import { Platforms } from "./platforms";
+import { Browsers } from "./browsers";
+import { Rows } from "./rows";
+import { Legend } from "./legend";
+import "./bcd.scss";
+
+const BROWSERS = {
+  "desktop": ["chrome", "edge", "firefox", "ie", "opera", "safari"],
+  "mobile": ["webview_android", "chrome_android", "firefox_android", "opera_android", "safari_ios", "samsunginternet_android", "edge_mobile"],
+  "server": ["nodejs"],
+  "webextensions-desktop": ["chrome", "edge", "firefox", "opera"],
+  "webextensions-mobile": ["firefox_android"]
+};
+
+export class BrowserCompatibilityTable extends Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      currentNoteId: null,
+      hasDeprecation: false,
+      hasExperimental: false,
+      hasNonStandard: false,
+      hasFlag: false,
+      hasPrefix: false,
+      hasNotes: false,
+      legendSet: false
+    }
+  }
+
+  gatherPlatformsAndBrowsers(document, category) {
+    let platforms = ["desktop", "mobile"];
+    let displayBrowsers = [...BROWSERS["desktop"], ...BROWSERS["mobile"]];
+    if (category === "javascript") {
+      displayBrowsers.push(...BROWSERS["server"]);
+      platforms.push("server");
+    }
+    if (category === "webextensions") {
+      displayBrowsers = [...BROWSERS["webextensions-desktop"], ...BROWSERS["webextensions-mobile"]];
+      platforms = ["webextensions-desktop", "webextensions-mobile"];
+    }
+    return [platforms, displayBrowsers];
+  }
+
+  onNotesClick = currentNoteId => {
+    if (this.state.currentNoteId && (this.state.currentNoteId === currentNoteId)) {
+      this.setState({ currentNoteId: null });
+    } else {
+      this.setState({ currentNoteId: currentNoteId });
+    }
```

Comment:
All of the above is a bit cryptic. The condition, can it not be expressed as:
```
onNotesClick = noteId => {
  this.setState({currentNoteId: noteId === this.state.currentNoteId ? null : noteId});
}
```

Reply:
Cleaner, I like it 👍